### PR TITLE
One keystroke cljs eval

### PR DESCRIPTION
**Pull request by cldwalker**
This is an attempt to make a clojurescript eval that just works for your project, much like clojure's eval and inspired by realizations in #8. Try it out:

~~EDIT: Not working with new clojurescript versions until #52 lands:~~

* Clone the Clojure plugin and checkout this branch.
* `lein new mies-om om-test`
* `cd om-test && lein cljsbuild once`
* Open LightTable and eval a line e.g. `(enable-console-print!)`
* An internal browser auto-opens and the line is evaled. You now have a cljs repl to your project (not LT's)!

The client auto-magically works by looking for a number of common places for index.html and opening it. We can make this a configurable behavior as well as disabling this feature entirely for power users. We could also make opening to an external browser a configurable behavior.

As for implementation, I resorted to a new connection as we may want to open an internal or external browser. We could reuse the [existing internal browser connection](https://github.com/LightTable/LightTable/blob/49f98f8ed8fc723be16a7f36111d8047d1c09e06/src/lt/objs/browser.cljs#L515) and same for the external one. However, that will require changing their :connects to kick off behaviorw and this feature will block until the next release. As for placement of the :connect call, I tried placing it in `:exec.cljs` behavior to less success. Current placement seems reasonable. If we don't like the tag check, we could change :create to point to a trigger e.g. `:create-eval-connection`

@LightTable/committers Would love to get feedback. I'm also going to open this up to the community on a mailing list thread.

### CODE REVIEW

**Comment by cldwalker** [comment]
Once #52 lands we'll have modern clojurescript eval and I'd like to make this a part of that release. @rundis @tonsky @kenny-evitt @stephenbrady Any thoughts on or objections to this?

File: src/lt/plugins/clojure.cljs [hunk 1]
```diff
@@ -870,3 +877,24 @@
...
 (def clj-lang (object/create ::langs.clj))
+
+(def browser-paths
+  "Relative paths to search for when connecting to a clojurescript browser."
+  ["index.html" "resources/public/index.html" "public/index.html" "dev-resources/public/index.html"])
+
+(defn find-default-html-file
+  "Searches browser-paths for first relative path to exist and returns it."
+  [ed]
+  (let [project-dir (files/parent (files/walk-up-find (get-in @ed [:info :path]) "project.clj"))]
```
Comment (stephenbrady) [inline src/lt/plugins/clojure.cljs]: This is all great. A little love for `build.boot` would be a nice touch.

**Comment by kenny-evitt** [comment]
@cldwalker No objections – only happy thoughts!

**Comment by tonsky** [comment]
Not sure why I’m in that list, but obviously I don’t mind :)

File: src/lt/plugins/clojure.cljs (hunk 1 above)
Comment (cldwalker) [inline src/lt/plugins/clojure.cljs]: Agreed but boot only works for browser external connections and this currently only supports browser (internal) connections. I'll add boot if we ever add browser external support for 'ClojureScript Browser'

**Comment by cldwalker** [comment]
Thanks for the feedback all. I've made both the default cljs client and the relative paths/urls to look for configurable via behaviors. I'll be releasing later today

**Comment by rundis** [comment]
:+1:

**Pull request merged**